src/DataFramesMeta.jl

```diff
@@ -44,6 +44,90 @@ replace_syms!(e::Expr, membernames) =
         mapexpr(x -> replace_syms!(x, membernames), e)
     end
 
+"""
+    @col(kw)
+
+`@col` transforms an expression of the form `z = :x + :y` into it's equivalent in
+DataFrames's `source => fun => destination` syntax.
+
+### Details
+
+Parsing follows the same convention as other DataFramesMeta macros, such as `@with`. All
+terms in the expression that are `Symbols` are treated as columns in the data frame, except
+`Symbol`s wrapped in `^`. To use a variable representing a column name, wrap the variable
+in `cols`.
+
+`@col` constructs an anonymous function based off the given expression. It then creates
+a `source => fun => destination` pair that is suitable for the `select`, `transform`, and
+`combine` functions in DataFrames.
+
+### Examples
+
+```julia
+julia> @col z = :x + :y
+[:x, :y] => (##595 => :z)
+```
+
+In the above example, `##595` is an anonymous function equivalent to the following
+
+```julia
+(_x, _y) -> _x + _y
+```
+
+```julia
+julia> df = DataFrame(x = [1, 2], y = [3, 4]);
+
+julia> DataFrames.transform(df, @col z = :x .* :y)
+2×3 DataFrame
+│ Row │ x     │ y     │ z     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 3     │ 3     │
+│ 2   │ 2     │ 4     │ 8     │
+
+
+```
+
+"""
+macro col(kw)
+    esc(fun_to_vec(kw))
+end
+
+# `combinetable` needs to be `true` when we have syntax of the form
+# `@based_on(gd, fun(:x, :y))` where `fun` returns a `table` object.
+# We don't create the "new name" pair because new names are given
+# by the table.
+function fun_to_vec(kw::Expr; combinetable = false)
+    if kw.head === :(=) || kw.head === :kw || combinetable
```

Comment:
what is the scenario for `:kw`?

Reply:
With a small tweak in the internals of `@byrow` we don't need to consider `:kw`. So we don't have to support it.

Reply:
Nevermind. The `:kw` path is actually important here for `@linq` to work. I am not sure why, however.

Reply:
See #169 https://github.com/JuliaData/DataFramesMeta.jl/issues/169